### PR TITLE
USHIFT-1810: Added argument parsing to launch_vm function in scenario.sh

### PR DESCRIPTION
**Pull request by vanhalenar**

### CODE REVIEW

File: test/bin/scenario.sh [hunk 1]
```diff
@@ -428,29 +428,66 @@ EOF
...
+            --fips_mode)
+                fips_mode=1
+                shift
+                ;;
+            --bootc_mode)
+                bootc_mode=1
+                shift
+                ;;
+            *)
+                record_junit "${vmname}" "vm-launch-args" "FAILED"
```
Comment (ggiguash) [inline test/bin/scenario.sh:484]: Ah, let's also add an error message telling us which option was not passed correctly?

File: test/bin/scenario.sh [hunk 2]
```diff
@@ -428,29 +428,66 @@ EOF
...
+                exit 1
+                ;;
+        esac
+    done
+
```
[the remark this answers is not in the document]
Comment (vanhalenar) [reply test/bin/scenario.sh:496]: Made `--vmname` optional, as suggested in another comment.

File: test/bin/scenario.sh [hunk 3]
```diff
@@ -428,29 +428,66 @@ EOF
...
+    local vm_disksize=20
+    local vm_nics=1
+    local fips_mode=0
+    local bootc_mode=0
+
+    while [ $# -gt 0 ]; do
+        case "$1" in
+            --vmname|--boot_blueprint|--network_name|--vm_vcpus|--vm_memory|--vm_disksize|--vm_nics)
+                var=${1/--/}
+                eval "${var}=\$2"
```
[the remark this answers is not in the document]
Comment (vanhalenar) [reply test/bin/scenario.sh]: Done

File: test/bin/scenario.sh [hunk 4]
```diff
@@ -428,29 +428,66 @@ EOF
...
+                shift 2
```
[the remark this answers is not in the document]
Comment (vanhalenar) [reply test/bin/scenario.sh]: Done

File: test/bin/scenario.sh [hunk 5]
```diff
@@ -428,29 +428,66 @@ EOF
...
-    local -r boot_blueprint="${2:-${DEFAULT_BOOT_BLUEPRINT}}"
-    local -r network_name="${3:-default}"
-    local -r vm_vcpus="${4:-2}"
-    local -r vm_memory="${5:-4096}"
-    local -r vm_disksize="${6:-20}"
-    local -r vm_nics="${7:-1}"
-    local -r fips_mode="${8:-0}"
-    local -r bootc_mode="${9:-0}"
+    # set defaults
+    local vmname
```
[the remark this answers is not in the document]
Comment (vanhalenar) [reply test/bin/scenario.sh]: Yes, sounds good, implemented

File: test/bin/scenario.sh (hunk 1 above)
Comment (vanhalenar) [reply test/bin/scenario.sh:491]: Done

File: test/bin/scenario.sh [hunk 6]
```diff
@@ -428,29 +428,73 @@ EOF
...
+                    error "Failed parsing arguments: ${var} value not set"
+                    record_junit "${vmname}" "vm-launch-args" "FAILED"
+                    exit 1
+                fi
+                ;;
+            --fips_mode)
+                fips_mode=1
+                shift
+                ;;
+            --bootc_mode)
```
Comment (pmtk) [inline test/bin/scenario.sh:485]: I'm thinking if we could drop `_mode` from these two? Just have `--fips` and `--bootc`

File: test/scenarios/periodics/el94-crel@el94-src@optional-upgrade.sh [hunk 7]
```diff
@@ -10,7 +10,7 @@ scenario_create_vms() {
         return 0
     fi
 	prepare_kickstart host1 kickstart.ks.template "${start_commit}"
-	launch_vm host1 rhel-9.4
+	launch_vm --boot_blueprint rhel-9.4
```
Comment (pmtk) [inline test/scenarios/periodics/el94-crel@el94-src@optional-upgrade.sh:13]:
```suggestion
	launch_vm
```

`rhel-9.4` is default value

File: test/scenarios/periodics/el94-src@fips-rpm.sh [hunk 8]
```diff
@@ -6,7 +6,7 @@ scenario_create_vms() {
     [[ "${UNAME_M}" =~ aarch64 ]] && { record_junit "setup" "scenario_create_vms" "SKIPPED"; exit 0; }
 
     prepare_kickstart host1 kickstart-liveimg.ks.template "" true
-    launch_vm host1 rhel-9.4-microshift-source-isolated "" "" "" "" "" "1"
+    launch_vm -microshift-source-isolated --fips
```
[the remark this answers is not in the document]
Comment (vanhalenar) [reply test/scenarios/periodics/el94-src@fips-rpm.sh]: fixed

File: test/scenarios/periodics/el94-src@fips.sh [hunk 9]
```diff
@@ -6,7 +6,7 @@ scenario_create_vms() {
     [[ "${UNAME_M}" =~ aarch64 ]] && { record_junit "setup" "scenario_create_vms" "SKIPPED"; exit 0; }
 
     prepare_kickstart host1 kickstart.ks.template rhel-9.4-microshift-source-isolated true
-    launch_vm host1 rhel-9.4-microshift-source-isolated "" "" "" "" "" "1"
+    launch_vm -microshift-source-isolated --fips
```
[the remark this answers is not in the document]
Comment (vanhalenar) [reply test/scenarios/periodics/el94-src@fips.sh]: fixed

File: test/scenarios/periodics/el94-src@offline.sh [hunk 10]
```diff
@@ -4,7 +4,7 @@
 
 scenario_create_vms() {
     prepare_kickstart host1 kickstart-offline.ks.template ""
-    launch_vm host1  rhel-9.4-microshift-source-isolated "" "" "" "" 0
+    launch_vm  -microshift-source-isolated --vm_nics 0
```
[the remark this answers is not in the document]
Comment (vanhalenar) [reply test/scenarios/periodics/el94-src@offline.sh]: fixed

File: test/scenarios/periodics/el94-src@rpm-standard1.sh [hunk 11]
```diff
@@ -4,7 +4,7 @@
 
 scenario_create_vms() {
     prepare_kickstart host1 kickstart-liveimg.ks.template ""
-    launch_vm host1 "rhel-9.4-microshift-source-isolated"
+    launch_vm -microshift-source-isolated
```
[the remark this answers is not in the document]
Comment (vanhalenar) [reply test/scenarios/periodics/el94-src@rpm-standard1.sh]: fixed

**Pull request merged**